### PR TITLE
purego: undefine Dlerror for Windows

### PR DESCRIPTION
**Pull request by hajimehoshi**
Let's wait for the discussion at #107

### CODE REVIEW

File: dlerror.go [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: 2023 The Ebitengine Authors
 
+//go:build darwin || linux
```
Comment (TotallyGamerJet) [inline dlerror.go:4]: This actually removes all platforms except Darwin and Linux. Shouldn't it be `!windows` to just remove windows.

**Review by TotallyGamerJet — COMMENTED** [review]
Obviously this is a backwards breaking change. How are we handling that?

Will this PR include the stubs or should I make that a separate PR

File: dlerror.go (hunk 1 above)
Comment (hajimehoshi) [reply dlerror.go:4]: Dl* is supported only by darwin or linux. If we want to use `!windows`, we should update all go:build for consistency. What do you think?

**Comment by hajimehoshi** [comment]
> Obviously this is a backwards breaking change. How are we handling that?

As purego is still in v0, I'm fine with this change. I guess nobody uses Dlerror for Windows.

> Will this PR include the stubs or should I make that a separate PR

As I'm against adding stubs, this PR doesn't aim to add any other stubs.

**Review by TotallyGamerJet — APPROVED** [review]
LGTM!

**Comment by hajimehoshi** [comment]
As I commented first, let's wait for the conclusion at #107 :-) (That's why this PR is draft)

**Comment by hajimehoshi** [comment]
OK we have reached an agreement :-)

**Pull request merged**